src/main.rs

```diff
@@ -1,3 +1,114 @@
+extern crate gio;
+extern crate gtk;
+
+use gio::prelude::*;
+use gio::MenuExt;
+use gtk::prelude::*;
+
+use std::env::args;
+
+macro_rules! upgrade_weak {
+    ($x:ident, $r:expr) => {{
+        match $x.upgrade() {
+            Some(o) => o,
+            None => return $r,
+        }
+    }};
+    ($x:ident) => {
+        upgrade_weak!($x, ())
+    };
+}
+
+fn build_menu(application: &gtk::Application, window: &gtk::ApplicationWindow) {
+    let menu = gio::Menu::new();
+    let menu_bar = gio::Menu::new();
+    let about_menu = gio::Menu::new();
+
+    menu.append("Quit", "app.quit");
+
+    about_menu.append("About", "app.about");
+    menu_bar.append_submenu("?", &about_menu);
+
+    // Connect the about menu entry click event.
+    let about = gio::SimpleAction::new("about", None);
```

Comment:
Modern GTK apps usually put this (and global preferences, etc) into the global per application menu, not inside the application. And inside the application there's usually a hamburger menu on the right for window-specific settings

Reply:
Never heard of it and my searches gave nothing. Is it a gtk/gio thing?

Reply:
I thought that's what the `gio::MenuItem` stuff is already doing but apparently not. If you run gnome-shell and open e.g. gedit, you'll have various of it's global menu items in the gnome-shell menu. And then various window-specific ones in the hamburger menu at the top-right (save as, find, etc.).

I'll research for the global menu tomorrow :)

Reply:
That's weird... Waiting more info on this. :D

Reply:
```diff
diff --git a/src/main.rs b/src/main.rs
index 00eea79..2992d52 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -21,13 +21,10 @@ macro_rules! upgrade_weak {
 
 fn build_menu(application: &gtk::Application, window: &gtk::ApplicationWindow) {
     let menu = gio::Menu::new();
-    let menu_bar = gio::Menu::new();
-    let about_menu = gio::Menu::new();
-
     menu.append("Quit", "app.quit");
 
-    about_menu.append("About", "app.about");
-    menu_bar.append_submenu("?", &about_menu);
+    let about_menu = gio::MenuItem::new("About", "app.about");
+    menu.append_item(&about_menu);
 
     // Connect the about menu entry click event.
     let about = gio::SimpleAction::new("about", None);
@@ -37,13 +34,15 @@ fn build_menu(application: &gtk::Application, window: &gtk::ApplicationWindow) {
         let p = gtk::AboutDialog::new();
         p.set_authors(&["Sebastian Dröge", "Guillaume Gomez"]);
         p.set_website_label(Some("github repository"));
-        p.set_website(Some("https://github.com/sdroege/rustfest-rome18-gtk-gst-workshop"));
+        p.set_website(Some(
+            "https://github.com/sdroege/rustfest-rome18-gtk-gst-workshop",
+        ));
         p.set_comments(Some("A process viewer GUI wrote with gtk-rs"));
         // p.set_copyright(Some("This is under MIT license"));
         p.set_transient_for(Some(&window));
+        p.set_modal(true);
         p.set_program_name("Rustfest GTK+GStreamer");
-        p.run();
-        p.destroy();
+        p.show_all();
     });
     let quit = gio::SimpleAction::new("quit", None);
     let weak_window = window.downgrade();
@@ -56,7 +55,6 @@ fn build_menu(application: &gtk::Application, window: &gtk::ApplicationWindow) {
     application.add_action(&quit);
 
     application.set_app_menu(&menu);
-    application.set_menubar(&menu_bar);
 }
 
 fn build_ui(application: &gtk::Application) {
```